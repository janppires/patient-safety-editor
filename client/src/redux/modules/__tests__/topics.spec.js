import topicsReducer, {
  addTopic,
  topicsSelector,
  topicSelector
} from "redux/modules/topics";

describe("reducer: topics.byId ", () => {
  it("adds new topic by id to empty state", () => {
    const topic = { id: 2, name: "random name" };
    const action = addTopic(topic);
    expect(topicsReducer({}, action)).toMatchSnapshot();
  });

  it("adds new topic by id to existing state", () => {
    const state = prepareState();
    const topic = { id: 2, name: "random name" };
    const action = addTopic(topic);
    expect(topicsReducer(state, action)).toMatchSnapshot();
  });

  const prepareState = () => ({
    byId: {
      33: {
        id: 33,
        name: "other 33 topic"
      }
    },
    list: [33]
  });
});

describe("topics selectors", () => {
  test("get topics array from state", () => {
    const state = prepareStateWithTopics();
    const topics = topicsSelector(state);
    expect(topics).toMatchSnapshot();
  });

  test("get undefined topic when topicId is unknown", () => {
    const state = prepareStateWithTopics();
    const topic = topicSelector(state);
    expect(topic).toMatchSnapshot();
  });

  test("get topic from id", () => {
    const state = prepareStateWithTopics();
    const topicId = 2;
    const topic = topicSelector(state, topicId);
    expect(topic).toMatchSnapshot();
  });

  const prepareStateWithTopics = () => ({
    topics: {
      list: [1, 2],
      byId: {
        1: { name: "random" },
        2: { name: "random 2" }
      }
    }
  });
});
