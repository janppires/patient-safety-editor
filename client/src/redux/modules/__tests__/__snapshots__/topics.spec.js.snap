// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer: topics.byId  adds new topic by id to empty state 1`] = `
Object {
  "byId": Object {
    "2": Object {
      "id": 2,
      "name": "random name",
    },
  },
  "list": Array [
    2,
  ],
}
`;

exports[`reducer: topics.byId  adds new topic by id to existing state 1`] = `
Object {
  "byId": Object {
    "2": Object {
      "id": 2,
      "name": "random name",
    },
    "33": Object {
      "id": 33,
      "name": "other 33 topic",
    },
  },
  "list": Array [
    33,
    2,
  ],
}
`;

exports[`topics selectors get topic from id 1`] = `
Object {
  "name": "random 2",
}
`;

exports[`topics selectors get topics array from state 1`] = `
Array [
  Object {
    "name": "random",
  },
  Object {
    "name": "random 2",
  },
]
`;

exports[`topics selectors get undefined topic when topicId is unknown 1`] = `undefined`;
